escrow/contracts/escrow.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Unlicensed
+pragma solidity ^0.8.9;
+
+import "@acala-network/contracts/token/Token.sol";
+
+contract Escrow {
+    address payable public requestor;
+    address payable public serviceProvider;
+    address public tokenAddress;
+    uint256 public amount;
+    ServiceStatus public requestorStatus;
+    ServiceStatus public serviceProviderStatus;
+
+    enum ServiceStatus {
+        Pending,
+        Confirmed,
+        Denied
+    }
+
+    constructor(
+        address _tokenAddress,
+        uint256 _amount,
+        address payable _requestor,
+        address payable _serviceProvider
+    ) {
+        requestor = _requestor;
+        serviceProvider = _serviceProvider;
+        tokenAddress = _tokenAddress;
+        amount = _amount;
+        requestorStatus = ServiceStatus.Pending;
+        serviceProviderStatus = ServiceStatus.Pending;
+    }
+
+    function requestorConfirmTaskCompletion(bool _taskConfirmation) public {
+        require(
+            msg.sender == requestor,
+            "Only the requestor can confirm his part"
+        );
+
+        if (_taskConfirmation == true) {
+            requestorStatus = ServiceStatus.Confirmed;
+        } else {
+            requestorStatus = ServiceStatus.Denied;
+        }
+
+        completeTask();
```

Comment:
If you add a successful execution feedback to the function, you can make sure it's successfully executed by changing this to:
```suggestion
        require(completeTask());
```